### DISCUSSION TITLE
shuffle not shuffling

### DISCUSSION THREAD

**Discussion by msmcknight** [Geektoolkit/Dynaframe3#121] [Q&A] [answered]
I have Dynaframe set to shuffle, but it's still displaying pictures in the alphabetical order of the filename.

When I start Dynaframe and it displays the playlist of files, they look randomized, and it also reports the line:
5/19/2021 2:02 AM:Shuffle: True

But once it gets going, the behavior is odd.  It's like it picks a random picture to start with, but then it's sequential by filename after that.  If I stop and start Dynaframe again, it does the same thing... randomly picks a starting file and then sequentially by name.  The starting file does appear to be random with each run, but that's the only random one in the entire playlist.

The filenames are fairly generic... they might be DSCN_XXXX.JPG or IMG_XXXX.JPG, where XXXX is a number between 0001 and 9999.

My appsettings.json file looks like this:
`{"Shuffle":true,"VideoVolume":false,"ExpandDirectoriesByDefault":false,"Rotation":0,"OXMOrientnation":"--orientation 0","SearchDirectories":["/home/pi/Pictures"],"CurrentPlayList":[],"InfoBarFontSize":50,"InfoBarState":5,"FadeTransitionTime":0,"SlideshowTransitionTime":10000,"DateTimeFormat":"H:mm tt","DateTimeFontFamily":"Terminal","CurrentDirectory":"/home/pi/Pictures","NumberOfSecondsToShowIP":0,"ImageStretch":2,"VideoStretch":"Letterbox","ReloadSettings":true,"RefreshDirctories":true,"ListenerPort":8000,"IsSyncEnabled":false,"RemoteClients":[]} `

I'm using v2.13 downloaded today (May 19, 2021)

Has anyone else seen this?  Any ideas on how to make it properly shuffle the files?

**Comment by msmcknight** [comment, marked as answer]
OK, so I stumbled across the 2.18 version and it appears to have fixed the shuffle problem.

For others not so familiar with github, you can find additional releases here:
https://github.com/Geektoolkit/Dynaframe3/releases

**Reply by Geektoolkit** [reply]
Yes sorry it was broken in previous builds and should be much better in 2.18. Please don't jump much further ahead things get rocky at 2.20

**Reply by msmcknight** [reply]
As a report on 2.18 so far.... I've had 450 images in the "shuffle" rotation for more than 24 hours and no crashes.  Seems to be running smooth.

**Reply by msmcknight** [reply]
Out of curiosity, what is the max number of pictures supported?  My picture library is well over 100,000.   If I could add all of them at 10s each, that would mean I wouldn't see the same picture again for almost a year!  Assuming of course that the "shuffler" wont show the same picture again until all pictures have been shown once.

**Reply by Geektoolkit** [reply]
Thanks for the report on 2.18! I may promote that to release. As far as max number...I don't know if there's a number really...it's kinda based on the limits of the raspberry pi that you're using.   know that we've seen stress tests of 10,000 images seem to work, but I don't know how long they ran for.
  The list code isn't super efficient, and honestly it's about to become alot less so if I add in the exif data. Right now I store the full path to every image in a playlist in memory...I am contemplating creating a database locally and storing it, and the exif data, in there. I want to play with understanding how much memory I'm looking at when I get to that scale. 
  Right now my focus is on 'blurboxing'.  If you have 100k images and they're not exactly the resolution of the monitor, then you're likely noticing that things either get cropped or they end up scaled off screen...blurboxing from my initial tests gives a good option to not scale off the screen (uniformtofill), maintain aspect ratio, but hides the black bars.  Once I get that working I'll look into exifdata and other methods of palylist management, as well as the showing exif data, sorting by date work.

**Reply by qwksilver** [reply]
I can confirm that a library of 120,000 will load on a pi4b-2gb, the time to first photo is close to 5 min, and it seems locked up while generating the shuffle data.  the shuffle data will reset each time the pi is rebooted, and the shuffledata may reset if it detects a new file added to the directory.  in my case the photos were stored on a machine other than the pi, via wireless network.  

the way the list forms you are correct, the list doesn't repeat photos till it reaches the end (or the whole list resets)
